clang/lib/Sema/SemaExpr.cpp

```diff
@@ -226,6 +226,13 @@ bool Sema::DiagnoseUseOfDecl(NamedDecl *D, ArrayRef<SourceLocation> Locs,
                !IsConst && VD->hasGlobalStorage() && !isa<ParmVarDecl>(VD))
         SYCLDiagIfDeviceCode(*Locs.begin(), diag::err_sycl_restrict)
             << Sema::KernelGlobalVariable;
+      // Using ESIMD globals in SYCL context is not allowed
+      else if (IsRuntimeEvaluated && isSYCLEsimdPrivateGlobal(VD) &&
+               VD->hasGlobalStorage() && !isa<ParmVarDecl>(VD) &&
+               !getCurFunctionDecl()->hasAttr<SYCLSimdAttr>())
```

Comment:
I think this can generate a null pointer dereference if the variable is outside of a function, probably should have test coverage for that.

Reply:
I couldn't come up with a test for this but added protection against a crash.